### PR TITLE
Replace `http_get` with `load_dataset` -`askubuntu` and `all-nli`

### PR DESCRIPTION
**Pull request by omkar-334**
~~Still have to replace `askubuntu` train - draft for now...~~
This Pull Request replaces legacy datasets - `askubuntu` and `all-nli` with the newer versions hosted on huggingface.

### CODE REVIEW

**Comment by omkar-334** [comment]
Hey @tomaarsen , this PR is ready for review..   
Here is the link for [Colab](https://colab.research.google.com/drive/1D7nZEO1N1STwQ94Qbrh9HRLawhDVWZO0?usp=sharing) where I've compared the old and new datasets.   

1. `Askubuntu`
Screenshot 2026-01-30 at 3 51 30 PM   


The new `train` split has significantly less samples than the old one (old has 160k, new has 27k samples. All 27k samples are included in the old one). The `test` and `dev` are almost same.

2. `All-NLI`
Screenshot 2026-01-30 at 3 52 01 PM   


Here all splits compared have the same number of samples.   
The only problem here is that the labels are different, for which I've added logic. Can you take a look at this and let me know?

Thanks!

**Comment by omkar-334** [comment]
I've also addressed your suggestions on the migration scripts, but I think this PR should be merged before them.

**Comment by tomaarsen** [comment]
> I've also addressed your suggestions on the migration scripts, but I think this PR should be merged before them.

Okay, I'll focus on this PR before the migration ones then! I'll look into the AskUbuntu issue.
For the dev and test, that's caused by this line: https://github.com/huggingface/sentence-transformers/blob/b07e0cebc43de12806ba8fe9de60468f84593647/examples/sentence_transformer/unsupervised_learning/TSDAE/train_askubuntu_tsdae.py#L65-L66

My uploaded version also includes the data that were skipped here. For matching results, you'll need:
```python
load_dataset("sentence-transformers/askubuntu", split="dev").filter(lambda x: len(x["positive"]) > 0)
# or this, which is identical but perhaps more confusing? It's fine either way if you have a comment
load_dataset("sentence-transformers/askubuntu", split="dev").filter(lambda x: x["positive"])
'''
Dataset({
    features: ['query', 'positive', 'negative'],
    num_rows: 189
})
'''
```

With https://huggingface.co/datasets/sentence-transformers/askubuntu-questions something indeed went wrong, not sure what. I've now uploaded the full corpus, so that includes the queries that are in the dev/test dataset(!). So:

```python
from datasets import load_dataset

askubuntu_dev = load_dataset("sentence-transformers/askubuntu", split="dev")
askubuntu_dev = askubuntu_dev.filter(lambda x: x["positive"])  # Skip samples without positive passages
print(askubuntu_dev)
"""
Dataset({
    features: ['query', 'positive', 'negative'],
    num_rows: 189
})
"""

askubuntu_test = load_dataset("sentence-transformers/askubuntu", split="test")
askubuntu_test = askubuntu_test.filter(lambda x: x["positive"])  # Skip samples without positive passages
print(askubuntu_test)
"""
Dataset({
    features: ['query', 'positive', 'negative'],
    num_rows: 186
})
"""

askubuntu_corpus = load_dataset("sentence-transformers/askubuntu-questions", split="train")
print(askubuntu_corpus)
"""
Dataset({
    features: ['text'],
    num_rows: 167765
})
"""
dev_or_test_questions = (
    set(askubuntu_dev["query"])
    | {question for positive_list in askubuntu_dev["positive"] for question in positive_list}
    | {question for negative_list in askubuntu_dev["negative"] for question in negative_list}
    | set(askubuntu_test["query"])
    | {question for positive_list in askubuntu_test["positive"] for question in positive_list}
    | {question for negative_list in askubuntu_test["negative"] for question in negative_list}
)
askubuntu_corpus = askubuntu_corpus.filter(lambda x: x["text"] not in dev_or_test_questions)
print(askubuntu_corpus)
"""
Dataset({
    features: ['text'],
    num_rows: 160425
})
"""
```
Looks like there's a tiny difference in the final filtered train, but I think it's probably some question that existed twice with different IDs, so that matching on IDs (the original script) doesn't work, but matching on questions themselves (like now) does. It seems fine to use this, though.

- Tom Aarsen

**Comment by omkar-334** [comment]
Thanks for this @tomaarsen ! `ask-ubuntu` looks good to go....  
A minor question for `ask-nli` - The labels in the original dataset and the labels in the hf dataset are different, so I wrote a mapping to convert them.  
Do you think this is needed?

**Comment by tomaarsen** [comment]
> A minor question for ask-nli - The labels in the original dataset and the labels in the hf dataset are different, so I wrote a mapping to convert them.

Ah, yes. This was likely an oversight of mine when I uploaded the dataset. I don't think the mapping is necessary, I'm not using one in other training scripts. You can remove it, if you'd like.

- Tom Aarsen

**Comment by omkar-334** [comment]
@tomaarsen thanks, this looks good now

File: examples/sentence_transformer/unsupervised_learning/CT/train_askubuntu_ct.py
Comment (tomaarsen) [inline examples/sentence_transformer/unsupervised_learning/CT/train_askubuntu_ct.py:1]: This file doesn't work now; the second half requires `torch.utils.data.Dataset` because it hasn't been updated with the v3 format yet.

File: examples/sentence_transformer/unsupervised_learning/CT_In-Batch_Negatives/train_askubuntu_ct-improved.py
Comment (tomaarsen) [inline examples/sentence_transformer/unsupervised_learning/CT_In-Batch_Negatives/train_askubuntu_ct-improved.py:1]: This file doesn't work now; the second half requires `torch.utils.data.Dataset` because it hasn't been updated with the v3 format yet. You have a PR here to help: https://github.com/huggingface/sentence-transformers/pull/3628

File: examples/sentence_transformer/unsupervised_learning/SimCSE/train_askubuntu_simcse.py
Comment (tomaarsen) [inline examples/sentence_transformer/unsupervised_learning/SimCSE/train_askubuntu_simcse.py:1]: This file doesn't work now; the second half requires `torch.utils.data.Dataset` because it hasn't been updated with the v3 format yet. You have a PR here to help: https://github.com/huggingface/sentence-transformers/pull/3630

File: examples/sentence_transformer/unsupervised_learning/CT/train_askubuntu_ct.py
Comment (omkar-334) [reply examples/sentence_transformer/unsupervised_learning/CT/train_askubuntu_ct.py:1]: I can make a PR for this

**Comment by omkar-334** [comment]
> There's 3 examples files using AskUbuntu that don't work if I merge this PR, but I think that's fine as there's follow-up PRs for 2 of the 3 files already to update it fully to v3. If you'd like to help pull those over the finish line, then I'm okay with merging this despite the example scripts (temporarily) not working.

Sure, I can write a PR for the 1 remaining file, and if anything else is broken after merging this and the other 2. I'll maybe have the update those 2 PRs

**Comment by tomaarsen** [comment]
Perfect, then we'll merge this when the tests pass. Thanks!

- Tom Aarsen

**Comment by omkar-334** [comment]
> Perfect, then we'll merge this when the tests pass. Thanks!
> 
> * Tom Aarsen

@tomaarsen I've updated the branch, do you need to approve the tests again? Sorry🥲

**Comment by tomaarsen** [comment]
I do, all good. There's some issues I've spotted, so I've got to push another commit in a second either way!

File: examples/sentence_transformer/unsupervised_learning/CT/train_askubuntu_ct.py
Comment (omkar-334) [reply examples/sentence_transformer/unsupervised_learning/CT/train_askubuntu_ct.py:1]: #3647
